std/os/file.zig

```diff
@@ -324,7 +324,7 @@ pub const File = struct {
                         else => os.unexpectedErrorPosix(err),
                     };
                 }
-                return result;
+                return @intCast(u64, result);
```

Comment:
Is this cast needed? I'm not aware of a target where the pointer size is more than 64 bits.

Reply:
32-bit platforms can perform I/O on files bigger than `0xFFFFFFFF`.

Reply:
All the `lseek` functions return `usize`. You can implicit cast `usize` to `u64`:

```zig
test "aeou" {
    var x: usize = 1234;
    var y: u64 = x;
}
```

I'm not aware of a target on which this doesn't work.

If it were `lseek64` it would definitely be a redundant cast - going from u64 to u64.

Reply:
Screw that, I added that cast because I was getting
`error: expected type 'std.os.file.GetSeekPosError!u64', found 'usize'`

Reply:
Ahh. In that case, you can help the type system along by introducing an intermediary implicit cast:
`return u64(result);`

It's preferable to do this because [Implicit casts are only allowed when it is completely unambiguous how to get from one type to another, and the transformation is guaranteed to be safe.](https://ziglang.org/documentation/master/#Implicit-Casts)

Reply:
Done!